src/creds.rs

```diff
@@ -209,32 +219,34 @@ impl CredentialBundle {
         credential_type: CredentialType,
         ciphersuite_name: CiphersuiteName,
     ) -> Result<Self, CredentialError> {
-        let keypair = Ciphersuite::new(ciphersuite_name).new_signature_keypair();
         let ciphersuite = Ciphersuite::new(ciphersuite_name);
+        let (private_key, public_key) = ciphersuite.new_signature_keypair().as_tuple();
         let mls_credential = match credential_type {
-            CredentialType::Basic => Ok(BasicCredential {
+            CredentialType::Basic => BasicCredential {
                 identity,
                 ciphersuite,
-                public_key: keypair.get_public_key().clone(),
-            }),
-            _ => Err(CredentialError::UnsupportedCredentialType),
-        }?;
+                public_key,
+            },
+            _ => return Err(CredentialError::UnsupportedCredentialType),
+        };
         let credential = Credential {
             credential_type,
             credential: MLSCredentialType::Basic(mls_credential),
         };
         Ok(CredentialBundle {
             credential,
-            signature_private_key: keypair.get_private_key().clone(),
+            signature_private_key: private_key,
         })
     }
 
-    /// Get a reference to the signature private key.
-    pub fn signature_private_key(&self) -> &SignaturePrivateKey {
-        &self.signature_private_key
-    }
-
     pub fn credential(&self) -> &Credential {
         &self.credential
     }
+
+    /// Sign a `msg` using the private key of the credential bundle.
+    pub(crate) fn sign(&self, msg: &[u8]) -> Result<Signature, SignatureError> {
```

Comment:
Hm, not sure if propagating the `SignatureError` up here is the right thing to do. It has to be handled somewhere. But I guess we can leave this to #83 .